### PR TITLE
Учебный проект: разделяй и властвуй

### PR DESCRIPTION
**Pull request by akzhar**
---
:mortar_board: [Учебный проект: разделяй и властвуй](https://up.htmlacademy.ru/javascript/17/user/490939/tasks/11)

:boom: https://htmlacademy-javascript.github.io/490939-code-and-magick-17/6/

### CODE REVIEW

File: js/setup.js [hunk 1]
```diff
@@ -1,164 +1,166 @@
...
+    'yellow',
+    'green'
+  ];
+  var fireBallColors = [
+    '#ee4830',
+    '#30a8ee',
+    '#5ce6c0',
+    '#e848d5',
+    '#e6e848'
+  ];
```
Comment (rkhaslarov) [inline js/setup.js]: I think, all mocked data should be moved to the separate module data.js, which will be removed in the future =)

File: js/setup.js [hunk 2]
```diff
@@ -1,164 +1,166 @@
...
+
+  function getRandomElemInArr(arr) {
```
Comment (rkhaslarov) [inline js/setup.js]: Kind of utils function, should be in the utils module =)

File: js/setup.js [hunk 3]
```diff
@@ -1,164 +1,166 @@
...
+    var wizards = [];
+    for (var i = 0; i < n; i++) {
+      wizards[i] = {
+        name: getRandomElemInArr(firstNames) + ' ' + getRandomElemInArr(lastNames),
+        coatColor: getRandomElemInArr(coatColors),
+        eyesColor: getRandomElemInArr(eyesColors)
+      };
+    }
+    return wizards;
+  }
```
Comment (rkhaslarov) [inline js/setup.js]: setup is too common for such functions, probably we should create wizard.js as well

File: js/setup.js [hunk 4]
```diff
@@ -1,164 +1,166 @@
...
+    inputFireBallColor.value = newFireBallColor;
+  }
+
+  var similarWizards = createWizards(NUMBER_OF_WIZARDS);
+  var fragment = document.createDocumentFragment();
+  for (var i = 0; i < similarWizards.length; i++) {
+    var wizardCard = renderWizard(similarWizards[i]);
+    fragment.appendChild(wizardCard);
+  }
+  similarWizardsList.appendChild(fragment);
```
Comment (rkhaslarov) [inline js/setup.js]: var similarWizards = createWizards(NUMBER_OF_WIZARDS);
similarWizardsList.appendChild(fragment);	  var fragment = document.createDocumentFragment();
similarWizardsBlock.classList.remove('hidden');	  for (var i = 0; i < similarWizards.length; i++) {

     var wizardCard = renderWizard(similarWizards[i]);
formOpenBtn.addEventListener('click', onFormOpenBtnClick);	    fragment.appendChild(wizardCard);
formOpenIcon.addEventListener('keydown', onFormOpenIconEnterPress);	  }
  similarWizardsList.appendChild(fragment);

looks like separate function =)

File: js/stats.js [hunk 5]
```diff
@@ -1,66 +1,68 @@
...
-  drawCloud(ctx, CLOUD_X, CLOUD_Y);
-  drawCloudHeader(ctx, CLOUD_X + 30, 20);
-  for (var i = 0; i < names.length; i++) {
-    var player = {
-      index: i,
-      name: names[i],
-      score: Math.round(times[i])
-    };
-    drawBar(ctx, player, scoreMax);
+  function getRandom(min, max) {
```
Comment (rkhaslarov) [inline js/stats.js]: utils

File: js/setup.js [hunk 6]
```diff
@@ -1,164 +1,166 @@
...
+  similarWizardsBlock.classList.remove('hidden');
+
+  formOpenBtn.addEventListener('click', onFormOpenBtnClick);
+  formOpenIcon.addEventListener('keydown', onFormOpenIconEnterPress);
```
Comment (rkhaslarov) [inline js/setup.js]: All necessary form functions should be moved to form js
Setup.js can be renamed to main.js
Main.js starts being the single entry point, which calls all necessary function from different modules

File: js/setup.js (hunk 1 above)
Comment (akzhar) [reply js/setup.js]: Done

File: js/setup.js (hunk 2 above)
Comment (akzhar) [reply js/setup.js]: Done

File: js/setup.js (hunk 3 above)
Comment (akzhar) [reply js/setup.js]: You' re right. Done

File: js/setup.js (hunk 4 above)
Comment (akzhar) [reply js/setup.js]: Done

File: js/stats.js (hunk 5 above)
Comment (akzhar) [reply js/stats.js]: Done

File: js/setup.js (hunk 6 above)
Comment (akzhar) [reply js/setup.js]: Ok, but we use IIFE, so I can't call all modules in main.js...
To reduce server queries I can merge all modules via posthtml plugin (gulp) and make a single js file main.js

Comment (akzhar) [reply js/setup.js]: main.js added

**Pull request merged**